janus_server/src/aggregator/aggregation_job_driver.rs

```diff
@@ -545,7 +546,7 @@ impl AggregationJobDriver {
         aggregation_job: AggregationJob<L, A>,
         stepped_aggregations: Vec<SteppedAggregation<L, A>>,
         mut report_aggregations_to_write: Vec<ReportAggregation<L, A>>,
```

Comment:
nit: should these arguments be `&[SteppedAggregation]` and `&[ReportAggregation]` for consistency with `prep_steps`?

Reply:
First, the easy one: `report_aggregations_to_write` should stay as a `Vec`, as it needs to be mutable -- `process_response_from_helper` effectively uses this parameter to allow the caller to specify some report aggregations to write, and it needs to be able to add its own to the list. (I think this level of coupling is OK since this is a private method, which only exists because the "aggregate init" & "aggregate continue" implementations both eventually fall into the same logic.)

Now, the harder one: I changed `stepped_aggregations` as you suggest. This required a few changes as previously we were passing the `Vec` to pass ownership of the included `SteppedAggregation`, and ownership was handy because we were mutating some of the internal fields of the stepped aggregation. However, the mutated internal state is a `ReportAggregation`; I hope to do a pass over `datastore.rs` and remove public fields from structural types similarly to what this PR does to `message.rs`, so soon mutating the `ReportAggregation` won't work anyway. And, other than that, IMO it's cleaner to avoid mutation when possible.

Note this required a couple of new clones, but I think this is an OK tradeoff.

Reply:
First, the easy one: `report_aggregations_to_write` should stay as a `Vec`, as it needs to be mutable -- `process_response_from_helper` effectively uses this parameter to allow the caller to specify some report aggregations to write, and it needs to be able to add its own to the list. (I think this level of coupling is OK since this is a private method, which only exists because the "aggregate init" & "aggregate continue" implementations both eventually fall into the same logic.)

Now, the harder one: I changed `stepped_aggregations` as you suggest. This required a few changes as previously we were passing the `Vec` to pass ownership of the included `SteppedAggregation`, and ownership was handy because we were mutating some of the internal fields of the stepped aggregation. However, the mutated internal state is a `ReportAggregation`; I hope to do a pass over `datastore.rs` and remove public fields from structural types similarly to what this PR does to `message.rs`, so soon mutating the `ReportAggregation` won't work anyway. And, other than that, IMO it's cleaner to avoid mutation when possible.

Note this required a couple of new clones, but I think this is an OK tradeoff.